wandb/sdk/launch/agent/agent.py

```diff
@@ -589,6 +589,21 @@ def _check_run_finished(
         try:
             run = job_tracker.run
             status = run.get_status().state
+
+            # Ensure run was not stopped intentionally
+            if status == "preempted":
```

Comment:
This is confusing, the run state isn't necessarily pre-empted here, we just assumed it was because the pod (or the job) vanished. But the run is only actually preempted if the pod vanished, and the run is still in a running state

Reply:
Agreed, "preempted" isn't a good name. I'm renaming to "disconnected" but I'm happy to take other suggestions.